#!/bin/python
PACKAGE = "cr_dai_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

autofocus_enum = gen.enum([ gen.const("OFF",      int_t, 0, ""),
            gen.const("AUTO",     int_t, 1, ""),
                  gen.const("MACRO",      int_t, 2, ""),
                  gen.const("CONTINOUS_VIDEO",      int_t, 3, ""),
                  gen.const("CONTINOUS_PICTURE",      int_t, 4, "")],
            "Decimation algorithm type.")
gen.add("autofocus_mode", int_t, 1,"Decimation mode",  0, 0, 4, edit_method=autofocus_enum)

gen.add("autofocus_startx", int_t, 2,"X coord of top left",  0, 0, 2048)
gen.add("autofocus_starty", int_t, 2,"Y coord of top left",  0, 0, 2048)
gen.add("autofocus_width", int_t, 2,"width of ROI",  2048, 0, 2048)
gen.add("autofocus_height", int_t, 2,"height of ROI",  2048, 0, 2048)
gen.add("autofocus_min", int_t, 2,"min af",  100, 0, 255)
gen.add("autofocus_max", int_t, 2,"min af",  180, 0, 255)
gen.add("manual_focus", int_t, 4,"Focus value",  135, 0, 255)

gen.add("autoexposure_lock", bool_t, 8,"Auto exposure lock mode enabled or disabled",  False)
gen.add("autoexposure_startx", int_t, 16,"X coord of top left",  0, 0, 2048)
gen.add("autoexposure_starty", int_t, 16,"Y coord of top left",  0, 0, 2048)
gen.add("autoexposure_width", int_t, 16,"width of ROI",  2048, 0, 2048)
gen.add("autoexposure_height", int_t, 16,"height of ROI",  2048, 0, 2048)

gen.add("autoexposure_compensation", int_t, 32,"AE comp",  0, -9, 9)
gen.add("contrast", int_t, 64,"",  0, -10, 10)
gen.add("brightness", int_t, 128,"",  0, -10, 10)
gen.add("saturation", int_t, 256,"",  0, -10, 10)
gen.add("sharpness", int_t, 512,"",  0, 0, 4)
gen.add("chroma_denoise", int_t, 1024,"",  0, 0, 4)

gen.add("manual_exposure", int_t, 2048,"",  -1, -1, 10000)
gen.add("manual_iso", int_t, 2048,"",  -1, -1, 1599)

exit(gen.generate(PACKAGE, "cr_dai_ros_msgs", "CameraControl"))